React/Nextjs Project Post Setup Steps.md

```diff
@@ -0,0 +1,679 @@
+## Next.js project post setup steps
+
+
+#### Project example
+
+[JS React Example project](https://github.com/infinum/JS-React-Example)
+
+
+### General setup
+
+Before jumping into sections below, read pages for folder structure, naming conventions and other to see how general structure and setup should be in the project.
+
+Do not skip sections when doing setup below since some configurations depend on previous ones.
+
+In each section there will be a noted version for core packages because at the time you are reading this some configurations might
+change by package authors especially if new version exists. This doesn't mean setup will not work and if you encounter changes
+please inform your team lead.
+
+This post setup was done for Next.js **v9.5.3**. 
+
+
+### Folder restructuring
+
+1. Delete `styles` folder in the project root. In the project remove all code that was used from that folder.
+1. Create `src` folder in the project root and move `pages` folder into `src`
+1. Create `tests` folder in the project root. In there create `e2e`, `integration` and `unit` folders.
+1. Create `bin` folder in the project root with `build.sh` file
+1. Create `.github` folder in the project root with `CODEOWNERS` file
+1. Update `README` file
+1. Add `src/utils` folder
+1. Add `components/ui` and `components/containers` folders in `src` folder
+
+Note: Git will not track empty folders
+
+
+### Linter, formatter and Git hooks
+
+##### Depends on sections: none
+
+| Package | Version |
+|---|---:|
+| eslint | 7.9.x |
+| prettier | 2.1.x |
+| @infinumjs/eslint-config-react | 2.0.x |
+| husky | 4.3.x |
+| lint-staged | 10.4.x |
+
+To add linter, formatter and Git hooks support do the following:
+1. Run command
+    ```bash
+    yarn add --dev eslint prettier @infinumjs/eslint-config-react husky lint-staged
+    ```
+    or
+    ```bash
+    npm install --save-dev eslint prettier @infinumjs/eslint-config-react husky lint-staged
+    ```
+   to install `eslint` linter, `prettier` formatter, Infinum `eslint` config, `husky` and `lint-staged` to run 
+   pre-commit hooks on code.
+1. In project root create `.editorconfig` file and add following content
+    ```text
+    root = true
+    
+    [*]
+    charset = utf-8
+    indent_style = tab
+    insert_final_newline = true
+    trim_trailing_whitespace = true
+    ```
+1. In project root create `.prettierrc.json` file and add following content
+    ```json
+    {
+        "$schema": "http://json.schemastore.org/prettierrc",
+        "printWidth": 120,
+        "endOfLine": "lf",
+        "useTabs": true,
+        "arrowParens": "always",
+        "quoteProps": "as-needed",
+        "bracketSpacing": true,
+        "singleQuote": true,
+        "semi": true,
+        "trailingComma": "es5",
+        "parser": "babel-ts"
+    }
+    ```
+1. In project root create `.eslintrc` file and add following content
+    ```json
+    {
+       "extends": "@infinumjs/eslint-config-react"
```

Comment:
this (and the installation step) should be updated with the typescript specific package